Feature - Get recurrent calendar events with CalendarView

janis-vitols opened the pull request:
Original `BaseFolder#items` method was returning all calendar items and only `recurring master`.
To get from Exchange also recurring items `CalendarView` with `StartDate ` & `EndDate` attributes should be provided.

Here is more info about `CalendarView`: https://msdn.microsoft.com/en-us/library/aa564515.aspx

These changes add new method `BaseFolder#items_from_calendar_view` to which user must provide necessary `CalendarView` parameters. Exchange server will calculate all items (also recurring) for provided start & end date and return them.

janis-vitols commented:
@ebeigarts I have no idea why TravisCI is trying to install `exchanger (0.1.3)`. At the beginning I thought maybe my version bumping lead to this fail, but reverting commit didn't fix that. Any support here would be appreciated :)

janis-vitols commented:
@ebeigarts it looks like there were some conflict/problems with older `ruby 1.9.3` & `bundler`. I have added this `travis.yml` which now uses `ruby 2.3.0` & `bundler 1.11.2`.

I'm now thinking, maybe some method naming should be changed to reflect Exchange `expand` method call. Will refactor this part.
Not sure if I will be able to cover this functionality with tests, I have a little bit blocked access to Exchange servers right now :disappointed:

ebeigarts commented:
Looks good, can you add some documentation to README on how to use this new method?

janis-vitols commented:
Yes I will update README and also do small cleanup on method naming - probably will use word `expand` in method, like `#expanded_items` - as by passing this `CalendarView` it calls Exchange `expand/expanded` method.

janis-vitols commented:
@ebeigarts I have refactored method naming and added documentation. Should I bump version to `0.1.4` or you will do that?

ebeigarts commented:
It would look cleaner if we could pass Date objects directly. I think it should be enough to add `key .., type: Time` and [to_xml should handle the conversion](https://github.com/ebeigarts/exchanger/blob/e22294dfe147ab22330ad32aa69dc23196d92ef6/lib/exchanger/field.rb#L112). I will bump the version myself.

janis-vitols commented:
Let's hope I got your idea right. I have added `Time` type for `CalendarView` attributes `start_date` and `end_date`. Also updated documentation by removing `#xmlschema` as it's done automatically now.

ebeigarts commented:
Released as v0.1.4
